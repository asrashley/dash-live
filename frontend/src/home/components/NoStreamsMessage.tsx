import { routeMap } from "@dashlive/routemap";
import { Card } from "../../components/Card";
import { Icon } from "../../components/Icon";

export function NoStreamsMessage() {
  return (
    <Card id="no-streams-msg" header="Please add some media files">
      <p>
        <Icon className="fs-2 me-2" name="camera-video-off" />
        There are no streams in the database. Please go to{" "}
        <a className="link" href={routeMap.listStreams.url()}>
          the streams page
        </a>{" "}
        and add some streams.
      </p>
    </Card>
  );
}
