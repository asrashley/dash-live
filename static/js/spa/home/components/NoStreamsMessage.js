import { html } from "htm/preact";
import { routeMap } from '/libs/routemap.js';
import { Icon, Card } from '@dashlive/ui';

export function NoStreamsMessage() {
    return html`<${Card} header="Please add some media files">
    <p><${Icon} className="fs-2 me-2" name="camera-video-off" />There are no streams in the database. Please go
    to <a className="link" href="${routeMap.listStreams.url()}">the streams page</a> and
    add some streams.</p>
    </${Card}>`;
}