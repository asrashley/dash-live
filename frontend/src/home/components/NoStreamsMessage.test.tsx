import { describe, expect, test } from "vitest";

import { renderWithProviders } from "../../test/renderWithProviders";
import { NoStreamsMessage } from "./NoStreamsMessage";

describe("NoStreamsMessage", () => {
  test("should display message", () => {
    const { asFragment, getByText } = renderWithProviders(
      <NoStreamsMessage />
    );
    getByText('Please add some media files');
    expect(asFragment()).toMatchSnapshot();
  });
});
