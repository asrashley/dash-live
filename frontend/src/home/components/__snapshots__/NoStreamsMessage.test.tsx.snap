// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoStreamsMessage > should display message 1`] = `
<DocumentFragment>
  <div
    class="card"
    id="no-streams-msg"
  >
    <div
      class="card-header"
    >
      Please add some media files
    </div>
    <div
      class="card-body"
    >
      <p>
        <span
          aria-label="camera-video-off icon"
          class="bi icon bi-camera-video-off fs-2 me-2"
          role="img"
        />
        There are no streams in the database. Please go to 
        <a
          class="link"
          href="/streams"
        >
          the streams page
        </a>
         and add some streams.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
